### PR TITLE
Expose Avi controller version in cluster config

### PR DESCRIPTION
**Pull request by chenlin07**
### What this PR does / why we need it

- Expose parameter `AVI_CONTROLLER_VERSION` for user to configure avi controller version
- Set default value of `AVI_CONTROLLER_VERSION` to `20.1.3`

We need to expose the parameter for supporting configurations of avi controller version, which is the new feature in akoo.
We need to set the default avi controller version, otherwise avi-sdk will pick the lowest version and cause invalid version error.
### Which issue(s) this PR fixes

Fixes # https://github.com/vmware-tanzu/tanzu-framework/issues/1746

### Describe testing done for PR

Manually e2e test:
Set the avi controller version to `20.1.1`, and apply the new cluster config. We can find the avi controller version is configured successfully in `ako-operator-data-values`
```
avi_labels: '""'
avi_disable_static_route_sync: true
avi_cni_plugin: antrea
avi_control_plane_ha_provider: true
avi_management_cluster_vip_network_name: VM Network
avi_management_cluster_vip_network_cidr: 10.206.208.0/20
avi_management_cluster_control_plane_vip_network_name: VM Network
avi_management_cluster_control_plane_vip_network_cidr: 10.206.208.0/20
avi_control_plane_endpoint_port: 6443
avi_controller_version: 20.1.1
```
### Release note

```release-note
Support avi controller version configuration
```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by XudongLiuHarold — COMMENTED** [review]
@chenlin07 , generally looks good! We need some test cases for this, you can add them here:https://github.com/vmware-tanzu/tanzu-framework/tree/main/pkg/v1/providers/tests/clustergen following the readme file.

**Review by XudongLiuHarold — APPROVED** [review]
LGTM, thanks!

**Review by lubronzhan — APPROVED** [review]

File: pkg/v1/providers/config_default.yaml [hunk 1]
```diff
@@ -553,6 +553,7 @@ AVI_USERNAME: ""
 AVI_PASSWORD: ""
 AVI_CLOUD_NAME:
 AVI_SERVICE_ENGINE_GROUP:
+AVI_CONTROLLER_VERSION: 20.1.3
```
Comment (vuil) [inline pkg/v1/providers/config_default.yaml:556]: It seems odd to have a default version value here. Pretty much everything specific to version values are driven by the BOM. Can you explain why this case is different?

Comment (lubronzhan) [reply pkg/v1/providers/config_default.yaml:556]: The AVI controller is a VM deployed by customer. This is the version to tell AKO to use sdk with this version to talk to the AVI Controller.
Customer could change this value if they change their AVI controller VM's version

**Comment by vuil** [comment]
lgtm, but we can use more details on the reason for the change and what tests were actually done.
The former is easiest done by filing an issue describing why the change is needed then reference the issue in the PR in the
'Fixes #' line. 
Also please add a short one liner in the release notes section about this new functionality as well.

File: pkg/v1/providers/config_default.yaml (hunk 1 above)
Comment (vuil) [reply pkg/v1/providers/config_default.yaml:556]: Thanks

**Comment by chenlin07** [comment]
Hi @vuil Thanks for reviewing the code! I have created the issue https://github.com/vmware-tanzu/tanzu-framework/issues/1746 and included more details in this PR.

**Pull request merged**